fix(22.04/libnss3): libs should contain all libraries

rebornplusplus opened the pull request:
This PR restructures the libnss3 slices. It renames the former `libs` slice to `nss`, to only include nss3-specific libraries. It also adds a new slice `smime` to include S/MIME libraries.

Finally, the `libs` slice is added back, which includes all of the slices that provide libraries. The rationale is that "libs" slices should contain all libraries coming from the package.

Note that the ca-certificates-java slice has also been updated to reflect this change.

Resolves #381

rebornplusplus commented on slices/libnss3.yaml:
```diff
@@ -1,18 +1,39 @@
     essential:
       - libc6_libs
       - libnspr4_libs
-      - libsqlite3-0_libs
```
Turns out we never needed `libsqlite3-0_libs` for the `libnss3.so` and `libnssutil3.so` files.

rebornplusplus replied:
Turns out I was wrong :facepalm: Added the slice back. Thanks @vpa1977.

rebornplusplus commented:
@vpa1977, will you please take a look at this PR? I could not add you to the reviewers list up there on the top-right.

vpa1977 commented on slices/libnss3.yaml:
```diff
@@ -23,6 +23,7 @@ slices:
+      - libsqlite3-0_libs
     contents:
       /usr/lib/*-linux-*/libnss3.so:
       /usr/lib/*-linux-*/libnssutil3.so:
```
We might want to add libsoftokn.

cjdcordeiro commented on slices/ca-certificates-java.yaml:
```diff
@@ -7,7 +7,7 @@ slices:
   data:
     essential:
       - ca-certificates_data
-      - libnss3_libs
```
I was about to complain about this change :p but it seems ok cause the contents are the same. At least for now. Are you sure `data` only depends on `nss` and not the other libraries?

rebornplusplus replied:
The `ca-certificates-java_data` slice currently have a `cacerts` file and an empty directory. We do not need the `nss` libraries at all **directly**, but they are used by the `openjdk` slices.

Anyway, yeah, it should be alright to include only the `nss` slice here.

cjdcordeiro replied:
Thanks for adding it back. And ok for the coming PR. Feel free to open it already

cjdcordeiro reviewed (approved):
It LGTM. There's a small comment for clarification. I'll also wait for @vpa1977 's review.

vpa1977 reviewed (approved):
We might want to add libsoftokn slice, but this can be done separately.

rebornplusplus commented:
> The `ca-certificates-java_data` slice currently have a `cacerts` file and an empty directory. We do not need the `nss` libraries at all **directly**, but they are used by the `openjdk` slices.

Hiya @vpa1977 and @cjdcordeiro, based on [this comment above](https://github.com/canonical/chisel-releases/pull/382#discussion_r1847952748), I have removed the `libnss3_nss` from `ca-certificates-java_data` and added it directly to `openjdk-8-jre-headless_security` slice. Please let me know what you think!

Also, I don't know if I was asleep or something before, but I finally added the `libsoftokn.so` files. :facepalm:

vpa1977 reviewed (changes requested):
I have added an integration test https://github.com/rebornplusplus/chisel-releases/pull/12 for the soft token and it is currently fails due to the provider error. 
Running it installing nss3 package normally succeeds. 

Maybe we should drop soft token changes and make a separate PR with the integration test?

rebornplusplus commented:
> Maybe we should drop soft token changes and make a separate PR with the integration test?

I agree. Let's add the soft token stuff in a later PR with passing tests.

cjdcordeiro commented:
Waiting for PRs for newer releases
